src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Build.Locator
+{
+    internal static class DotNetSdkLocationHelper
+    {
+        private static readonly Regex DotNetBasePathRegex = new Regex("Base Path:(.*)$", RegexOptions.Compiled | RegexOptions.Multiline);
```

Comment:
Is it worth compiling this regex? My intuition is that the process creation will dwarf the cost.

This whole thing is generally expected to be called once per caller process / app domain, right?